packages/renderer/MdToHtml.ts

```diff
@@ -582,8 +582,9 @@ export default class MdToHtml {
 		let cssStrings = noteStyle(options.theme, {
 			contentMaxWidth: options.contentMaxWidth,
 		});
-
-		let output = { ...this.allProcessedAssets(allRules, options.theme, options.codeTheme) };
+		// Deep copy of the returned object into output variable is required so that changes done to output variable
+		// doesn't get reflected on the cached entires of `this.allProcessedAssets_` property.
+		let output = JSON.parse(JSON.stringify(this.allProcessedAssets(allRules, options.theme, options.codeTheme)));
```

Comment:
I don't get what this does, but we really don't want this kind of things in the codebase. Also your message implies that the output of this function is modified by calling code, but that shouldn't happen. Something needs to be fixed properly somewhere

Reply:
Thanks @deepampriyadarshi for the explanation. The problem is that `outputAssetsToExternalAssets_` is modifying the input - and that should not happen because all functions should ideally be idempotent and return a new object.

So any chance you could tweak this function so that it doesn't change the input parameter? I think just `newOutput = { ...output, pluginAssets: output.slice() }` should get you there?

Reply:
> The problem is that outputAssetsToExternalAssets_ is modifying the input - and that should not happen because all functions should ideally be idempotent and return a new object.

Yes that's beacuse in general, object and arrays are passed by reference (more technically [Call by sharing](https://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_sharing)) and not by value, unlike primitive data types. Therefore we need to make separate `Deep-Copy` of objects which was not happening earlier in `let output = { ...this.allProcessedAssets(allRules, options.theme, options.codeTheme) };`

### Clarifications
So do you want me to keep this => `let output = JSON.parse(JSON.stringify(this.allProcessedAssets(allRules, options.theme, options.codeTheme)));` as well as modify `outputAssetsToExternalAssets_`
&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;OR
Just modify `outputAssetsToExternalAssets_`?

Reply:
Yes modify outputAssetsToExternalAssets_. If you do as I've shown above, by creating a copy of the relevant properties, it should work fine

Reply:
Done.. All the expected functionalities are retained, nothing broke... Hopefully it resolves the issue

Reply:
Yes hopefully that should be fine. Duplicating objects using spread and slice is the preferred way because most likely faster than JSON parse/serialize and it feels more proper too. You'll see that a lot in the reducers for instance.

Reply:
> Duplicating objects using spread and slice is the preferred way because most likely faster than JSON parse/serialize and it feels more proper too. You'll see that a lot in the reducers for instance.

Totally agreed. I was least bothered about performance. Was more focused on bringing up this cache inconsistency to your notice.